pyopenms_viz/_matplotlib/core.py

```diff
@@ -53,32 +63,23 @@ def _load_extension(self):
     def _create_figure(self):
         """
         Create a figure and axes objects,
-        for consistency with other backends, the self.fig object stores the matplotlib axes object
+        for consistency with other backends, the fig object stores the matplotlib axes object
         """
-        if (
-            self.fig is None
-            and self.width is not None
-            and self.height is not None
-            and not self.plot_3d
-        ):
-            self.superFig, self.fig = plt.subplots(
+        # TODO why is self.heigh and self.width checked if no alternatives
+        if self.width is not None and self.height is not None and not self.plot_3d:
+            self.fig, self.ax = plt.subplots(
                 figsize=(self.width / 100, self.height / 100), dpi=100
             )
-            self.fig.set_title(self.title)
-            self.fig.set_xlabel(self.xlabel)
-            self.fig.set_ylabel(self.ylabel)
-        elif (
-            self.fig is None
-            and self.width is not None
-            and self.height is not None
-            and self.plot_3d
-        ):
-            self.superFig = plt.figure(
+            self.ax.set_title(self.title)
+            self.ax.set_xlabel(self.xlabel)
+            self.ax.set_ylabel(self.ylabel)
+        elif self.width is not None and self.height is not None and self.plot_3d:
+            self.fig = plt.figure(
                 figsize=(self.width / 100, self.height / 100), layout="constrained"
             )
-            self.fig = self.superFig.add_subplot(111, projection="3d")
-            self.fig.set_title(self.title)
-            self.fig.set_xlabel(
+            self.ax = self.fig.add_subplot(111, projection="3d")
+            self.ax.set_title(self.title)
+            self.ax.set_xlabel(
```

Comment:
Can we add some of the aesthetic params (i.e. fontsize, which are currently hardcoded) to the config?

Reply:
This is kind of hard currently because the 3D projection than the 2D. Long term solution would be to separate 3D plots into their own classes.

Reply:
Will do this eventually but maybe in another PR